Packages/Networking/Sources/Networking/NetworkSession+Imgur.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+import Networkable
+
+public extension NetworkSession {
+    static var imgur: NetworkSession {
+        buildSession(urlString: "https://api.imgur.com/3/")
+    }
+
+    private static func buildSession(urlString: String) -> NetworkSession {
+        let baseURL = URL(string: urlString)
+        let requestBuilder = URLRequestBuilder(baseURL: baseURL)
+        let logging = LoggingMiddleware(type: .info)
+        let statusCodeValidation = StatusCodeValidationMiddleware()
+        let middlewares: [Middleware] = [
+            logging,
+            ErrorDecoderMiddleware<ImgurError>(),
+            statusCodeValidation
+        ]
+        let result = NetworkSession(requestBuilder: requestBuilder, middlewares: middlewares)
+        return result
+    }
+}
```

Comment:
Question: It is nearly similar to `NetworkSession+GitHub` and maybe every URL session in the future, just different in error middle type. Can we refactor the `buildSession(urlString:)` to support other sessions?